### PR TITLE
Use data attributes to 'sanity-test' the time series charts

### PR DESCRIPTION
**Pull request by david-mears-2**
Add data attributes to the containers of the time series and of the costs pie chart; put the rendered data in there; spot-check it for time series in e2e tests, and check everything for costs pie (small amount of data).

Using 50% +/- tolerance to accommodate future model changes that alter the precise numbers, this is a 'sense-check' to see if the numbers are in the expected order of magnitude.

### CODE REVIEW

File: nuxt.config.ts [hunk 1]
```diff
@@ -61,4 +61,6 @@ export default defineNuxtConfig({
       rApiBase: "", // https://nuxt.com/docs/getting-started/testing#registerendpoint
     },
   },
+
+  compatibilityDate: "2024-12-11",
```
Comment (david-mears-2) [inline nuxt.config.ts:65]: I'm mainly adding this in to silence the warning about it that happens on starting up the dev server, on the theory that not doing so causes it to be added dynamically, which possibly triggers a full app re-load on start-up (as updating this config normally does), which would slow down the development experience.

Docs: https://nuxt.com/docs/api/nuxt-config#compatibilitydate

File: components/CostsPie.client.vue [hunk 2]
```diff
@@ -5,6 +5,7 @@
       :id="chartContainerId"
       ref="chartContainer"
       :class="[props.hideTooltips ? hideTooltipsClassName : '']"
+      :data-test="JSON.stringify(costsData)"
```
Comment (EmmaLRussell) [inline components/CostsPie.client.vue:8]: Maybe call this something a bit less generic - `data-summary`?

**Review by EmmaLRussell — COMMENTED** [review]
Works a treat!

I don't really love stringfying JSON into HTML attributes, but I think that just an aesthetic squeamishness on my part. An alternative approach would be the one we took in WODIN where we had a separate summary component which would encode individual summary values into separate attributes. But you would need two of those, one for time series and one for (each) costs value. Probably not worth it. 

Maybe include expected number of points for time series as well as first and last?

**Comment by EmmaLRussell** [comment]
> I suppose there's a chance there are missing data points, but that seems like an unlikely scenario?

Maybe, depends how it's implemented. I couldn't remember if the day (x) value came from the data returned by the backend, or if the front end derives it from the index. Less likely in the latter case! (but perhaps the implementation might change... )

**Comment by david-mears-2** [comment]
> Maybe, depends how it's implemented. I couldn't remember if the day (x) value came from the data returned by the backend, or if the front end derives it from the index. Less likely in the latter case! (but perhaps the implementation might change... )

It's derived by index in the front end. We have to draw the line somewhere with sanity-checking the data, and I think this is a decent place to draw it.

**Comment by david-mears-2** [comment]
> It's derived by index in the front end. We have to draw the line somewhere with sanity-checking the data, and I think this is a decent place to draw it.

I changed my mind and added in a test for the number of data points since it doesn't take long to implement.

**Review by EmmaLRussell — APPROVED** [review]
Did you get the e2e tests working locally?

**Comment by david-mears-2** [comment]
> Did you get the e2e tests working locally?

No, unfortunately. It was on webkit and mobile safari browsers.

**Pull request merged**